vm/src/builtins/memory.rs

```diff
@@ -118,6 +119,7 @@ pub struct PyMemoryView {
     step: isize,
     exports: AtomicCell<usize>,
     format_spec: FormatSpec,
+    hash: OnceCell<PyHash>,
```

Comment:
is this needed to be cached?

Reply:
CPython cache the hash value, also because CPython allows to get the hash value after the memoryview is released if it has been hashed before.

Reply:
No, in memoryview it not always looks contiguous. Only this function GetContiguous will create a new PyBytes inside if the buffer is non-contiguous. If you create memoryview via FromBuffer, than the contiguous is depends on buffer.